platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/Models.kt

```diff
@@ -10,15 +10,34 @@ package io.bitdrift.capture
 /**
  * A monad for modeling success or failure operations in the Capture SDK.
  */
-typealias CaptureResult<V> = com.github.michaelbull.result.Result<V, Error>
+sealed class CaptureResult<out V> {
```

Comment:
you should take a look at the library we used, it can be helpful as a reference implementation for things like this:
> The Result type is modelled as an inline value class. This achieves zero object allocations on the happy path.

Reply:
I'm not saying we have to do it that way but it's a neat property of the library